Program.cs

```diff
@@ -1,7 +1,54 @@
 ﻿namespace Training {
+   /// <summary>Program that takes input, sorts characters, and adds a special character.</summary>
    internal class Program {
-      static void Main (string[] args) {
-         Console.WriteLine ("Hello, World!");
+      /// <summary>The entry point of the program.</summary>
+      static void Main () {
+         string input = GetInput ("Enter the input characters: ");
+         char specialChar = GetSpecialCharacter ();
+         bool isDescending = IsDescending ();
+         string result = SortAndAddSpecialCharToEnd (input.ToCharArray (), specialChar, isDescending);
+         Console.WriteLine ($"Sorted and moved special character: {result}");
+
+         /// <summary>Prompts the user for input and validates that it contains only letters.</summary>
+         /// <param name="prompt">The prompt message.</param>
+         /// <returns>The user-provided input containing only letters.</returns>
+         static string GetInput (string prompt) {
+            string input;
+            while (true) {
+               Console.Write ($"{prompt}");
+               input = Console.ReadLine ();
+               if (!input.Any (char.IsDigit) && input.Length > 0) return input;
+               Console.WriteLine ("Input should contain only letters. Please try again.");
+            }
+         }
+
+         ///<summary>Prompts the user to choose whether to sort the array in descending order.</summary>

```

Comment:
```suggestion
         /// <summary>Prompts the user to choose whether to sort the array in descending order.</summary>

```
@gdineshkumar219 Watch the space